### PR TITLE
[Task] UI Profile 

### PR DESCRIPTION
**Pull request by ghost**
## What's this PR do ?

- Implement function A
- Fix bugs on function B

## Checklist

- [ ] Write/Update Unit Test (if have)
- [ ] Fill link PR into ticket and the opposite
- [ ] Validate UI/Model/API with latest specs
- [ ] Add UI pics to this PR (if have).

## Library

_(List library, third party add new, include version)_

- Library: AppCompat v1.2.0

## Impacted Areas in Application

_(List features, api, models or services that this PR will affect)_

- foo/bar

## Demo

https://github.com/namnh-0652/flutter_movie_clean/assets/76760001/b2bf4369-58f9-4f3b-8a27-3c23331fb6ba

https://github.com/namnh-0652/flutter_movie_clean/assets/76760001/86a102ea-0714-493d-8a07-019dfa019c7d

### CODE REVIEW

File: lib/pages/login/login.dart [hunk 1]
```diff
@@ -135,7 +136,9 @@ class _LoginPageState extends State<LoginPage> {
           if (_formKey.currentState!.validate())
             {
               // TODO: call login here
```
Comment (namnh-0652) [inline lib/pages/login/login.dart]: remove this TODO

File: lib/pages/proflile/create_password/account_create_password_page.dart [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+import 'package:flutter_movie_clean/shared/utils/validate_helper.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:go_router/go_router.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+class AccountCreatePasswordPage extends StatefulWidget {
+  const AccountCreatePasswordPage({this.imagePath, this.username, Key? key})
+      : super(key: key);
+
+  static const String routeLocation = "/create_password";
```
Comment (namnh-0652) [inline lib/pages/proflile/create_password/account_create_password_page.dart]: make it consistent, `createPassword`

File: lib/pages/proflile/create_password/account_create_password_page.dart [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  bool _isEnableBtn = false;
+  bool _isObsecureText = true;
+
+  void _updateBtnState() {
+    setState(() {
+      _isEnableBtn = ValidateHelper.validatePassword(
+            context,
+            _passwordTextController.text,
+          ) ==
+          null;
```
Comment (namnh-0652) [inline lib/pages/proflile/create_password/account_create_password_page.dart:42]: is this format broken?

File: lib/pages/proflile/create_pin/account_create_pin_page.dart [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  void _updateBtnState() {
+    if (pin.length == 4) {
+      setState(() {
+        _isEnableBtn = pin.every((element) => element.isNotEmpty);
+      });
+    }
+  }
+
+  @override
+  void initState() {
```
Comment (namnh-0652) [inline lib/pages/proflile/create_pin/account_create_pin_page.dart]: remove

File: lib/pages/proflile/create_pin/account_create_pin_page.dart [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+class _AccountCreatePinPageState extends State<AccountCreatePinPage>
+    with WidgetsBindingObserver {
+  bool isKeyboardVisible = false;
+
+  bool _isEnableBtn = false;
+  final pin = ["", "", "", ""];
+
+  void _updateBtnState() {
+    if (pin.length == 4) {
```
Comment (namnh-0652) [inline lib/pages/proflile/create_pin/account_create_pin_page.dart]: is this redundant?

File: lib/pages/proflile/create_avatar/account_create_avatar_page.dart [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                top: offsetSelected?.dy ?? 0,
+                curve: Curves.easeInOutBack,
+                child: Container(
+                  decoration: BoxDecoration(boxShadow: [
+                    BoxShadow(
+                      color: AppColors.white.withOpacity(0.25),
+                      blurRadius: 10,
+                      offset: Offset(5.w, 5.h),
+                    )
+                  ], borderRadius: const BorderRadius.all(Radius.circular(50))),
```
Comment (truyentd-2427) [inline lib/pages/proflile/create_avatar/account_create_avatar_page.dart]: format code
```suggestion
                  ], borderRadius: const BorderRadius.all(Radius.circular(50)),),
```

File: lib/pages/proflile/create_password/account_create_password_page.dart [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+
+class _AccountCreatePasswordPageState extends State<AccountCreatePasswordPage>
+    with WidgetsBindingObserver {
+  final TextEditingController _passwordTextController = TextEditingController();
+  bool isKeyboardVisible = false;
+
+  bool _isEnableBtn = false;
+  bool _isObsecureText = true;
+
+  void _updateBtnState() {
```
Comment (truyentd-2427) [inline lib/pages/proflile/create_password/account_create_password_page.dart:36]: should be rename
```suggestion
  void _onPasswordChanged() {
```

File: lib/pages/proflile/create_usename/profile_create_username_page.dart [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+        onPressed: _isEnableBtn
+            ? () => context.pushNamed(
+                  AccountCreatePasswordPage.routeName,
+                  extra: {
+                    "imagePath": widget.imagePath,
+                    "username": _usernameTextController.text,
+                  },
+                )
+            : null,
+      ),
```
Comment (truyentd-2427) [inline lib/pages/proflile/create_usename/profile_create_username_page.dart:97]: i think the code looks better when we define like below
```suggestion
        onPressed: () {
             if (!_isEnableBtn) return;
             context.pushNamed(
                  AccountCreatePasswordPage.routeName,
                  extra: {
                    "imagePath": widget.imagePath,
                    "username": _usernameTextController.text,
                  },
                )
        }
      ),
```

File: lib/route/app_router.dart [hunk 9]
```diff
@@ -47,5 +54,59 @@ final appRouter = GoRouter(
         ),
       ],
     ),
+    // /Account/ChooseAvatar
+    // /
```
Comment (truyentd-2427) [inline lib/route/app_router.dart]: remove comment
```suggestion
    // /Account/ChooseAvatar
    // /
```

File: lib/pages/proflile/create_avatar/account_create_avatar_page.dart [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+        child: Column(
+          mainAxisSize: MainAxisSize.max,
+          children: [
+            SizedBox(height: 32.h),
+            Assets.images.icLogo.svg(),
+            SizedBox(height: 11.h),
+            _buildTextDescription(),
+            const SizedBox(height: 20),
+            _buildContentApp(),
+            const SizedBox(height: 28),
```
Comment (truyentd-2427) [inline lib/pages/proflile/create_avatar/account_create_avatar_page.dart]: missing multiscreen configs
```suggestion
            const SizedBox(height: 20.h),
            _buildContentApp(),
            const SizedBox(height: 28.h),
```

File: lib/pages/proflile/create_avatar/account_create_avatar_page.dart [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+      width: 1.sw - 60.w,
+      height: 50.h,
+      title: context.l10n.looksGood,
+      onPressed: _itemSelectedIndex != -1
+          ? () {
+              context.pushNamed(ProfileCreateUserPage.routeName, extra: {
+                "imagePath": _imageProfiles[_itemSelectedIndex].path
+              });
+            }
+          : null,
```
Comment (truyentd-2427) [inline lib/pages/proflile/create_avatar/account_create_avatar_page.dart:74]: like the comment above

File: lib/pages/proflile/create_password/account_create_password_page.dart [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+            _isEnableBtn ? AppColors.crimsonApprox : AppColors.black,
+        width: 1.sw,
+        height: 50.h,
+        title: context.l10n.looksStrong,
+        onPressed: _isEnableBtn
+            ? () => context.pushNamed(AccountCreatePinPage.routeName, extra: {
+                  "imagePath": widget.imagePath,
+                  "username": widget.username
+                })
+            : null,
```
Comment (truyentd-2427) [inline lib/pages/proflile/create_password/account_create_password_page.dart:108]: like above
```suggestion
        onPressed: _isEnableBtn
            ? () => context.pushNamed(AccountCreatePinPage.routeName, extra: {
                  "imagePath": widget.imagePath,
                  "username": widget.username
                })
            : null,
```

File: lib/pages/proflile/create_pin/account_create_pin_page.dart [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+      child: Form(
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: pin
+              .mapIndexed((index, element) => _buildPinInput(index))
+              .toList(),
+        ),
+      ),
+    );
+  }
```
Comment (truyentd-2427) [inline lib/pages/proflile/create_pin/account_create_pin_page.dart]: I think pin widget have complex logics, in this case we should extract it to specific class

File: lib/pages/proflile/create_usename/profile_create_username_page.dart (hunk 8 above)
Comment (ghost) [reply lib/pages/proflile/create_usename/profile_create_username_page.dart:97]: set `onPressed = null` in here support for no animation when user clicked button .

**Review by namnh-0652 — APPROVED** [review]

**Pull request merged**